### ISSUE TITLE
May you show how to create a Latex spec file?

### ISSUE THREAD

**Issue by sntgluca** [krassowski/jupyterlab-lsp#287]
## Description

I tried unsuccessfully to create a spec file for https://github.com/latex-lsp/texlab, to add Latex LSP support.

No matter what I try, the Latex lsp server (as indicated in the LSP status bar) is always missing.
I have installed `texlab`, and added a simple spec such as the following (in a few variations)

```python
{
  "LanguageServerManager": {
    "language_servers": {
      "latex": {
        "languages": ["latex"],
        "argv": ["/path/to/texlab", "--stdio"]
      }
    }
  }
}
```

I added it to `</path/to/environment>/etc/jupyter/jupyter_notebook_config.d/latex-lsp.json`

I tried with the `Advanced settings` in JupyterLab.

Nothing seems to help.
This is more of a POC for me, but I would really like to understand how things are working and why I am not able to control the settings.

Thanks for any help!

**Comment by bollwyvl** [comment]
Funny you mention this... I _might_ be close to getting texlab packaged for
conda-forge (windows takes a long time):

>
https://github.com/conda-forge/staged-recipes/pull/12029

Even if you don't want to install it that way, it will help if we can find
one configuration that works which we can "semi officially" support, with
docs and tests on multiple platforms, and keep up-to-date.

Once that's in, I'll be able to open a pr here!

**Comment by sntgluca** [comment]
Actually having a conda package would be my favourite choice, I am waiting for the follow up!

Creating new specs remains a black box for me - of the worse kind, the one I am not able to make work - but I think time will fix this...

**Comment by bollwyvl** [comment]
Sounds good: ran into the not-so-fun task of digging up all the licenses
for the statically linked rust libraries, but making progress...

**Comment by bollwyvl** [comment]
The texlab pr got merged, should have the builds up relatively soon.

>
Kind of a slammed week, so we'll see when I get up to it...

**Comment by bollwyvl** [comment]
Packages are up (will propagate to all the CDNs within the hour):

https://anaconda.org/conda-forge/texlab/files?version=2.2.0

**Comment by bollwyvl** [comment]
@sntgluca here's where I'm working on it: 

- pr: https://github.com/krassowski/jupyterlab-lsp/pull/288
- binder: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/bollwyvl/jupyterlab-lsp/add-texlab-spec?urlpath=lab%2Ftree%2Fatest%2Fexamples%2Fexample.tex)

_the binder is building, but takes _forever_ because of the `r` stuff, but it _should_ work..._

Turns out, being written in rust, `texlab` is extremely picky about the data it accepts... we had a protocol bug in `initialized` (had not yet backported from #278) so it would have been hard to get it working without change/compiling the frontend.

If you have specific use cases you would like to see working, please let us know here (or on the PR)... the [excellent docs](https://texlab.netlify.app/docs) seem pretty involved (_what?! something involving latex is complicated?!_), and things like forward/reverse search might be out of scope, requiring a separate extension.

**Comment by bollwyvl** [comment]
This was released as part of [v1.1.2](https://github.com/krassowski/jupyterlab-lsp/releases/tag/v1.1.2). The documentation has been [updated as well](https://jupyterlab-lsp.readthedocs.io/en/latest/Language%20Servers.html#Other-Scientific-Languages), so I think we can close this! Please reopen if you run into any issues!
